scripts/vote_2024_xx_xx_upgrade_optimism_bridge.py

```diff
@@ -0,0 +1,156 @@
+"""
+Voting xx/xx/2024.
+
+Upgrade L1Bridge, L2Bridge, L2 wstETH, L2 stETH, TokenRateOracle
+
+"""
+
+import time
+from brownie import interface
+from typing import Dict
+from brownie.network.transaction import TransactionReceipt
+from utils.voting import bake_vote_items, confirm_vote_script, create_vote
+from utils.ipfs import upload_vote_ipfs_description, calculate_vote_ipfs_description
+from utils.config import (
+    get_deployer_account,
+    get_is_live,
+    get_priority_fee,
+    network_name,
+)
+from utils.agent import agent_forward, agent_execute
+import eth_abi
+
+
+description = """
+
+Upgrade L1Bridge, L2Bridge, L2 wstETH, L2 stETH, TokenRateOracle
+
+"""
+
+OPTIMISM_BRIDGE_EXECUTOR: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+OPTIMISM_L1_L2_MESSAGE_SERVICE: str = "0x50c7d3e7f7c656493D1D76aaa1a836CedfCBB16A"
+PROXY_ADMIN_ADDR: str = "0xc6cdc2839378d50e03c9737723d96d117b09bda5"
+
+# 1. L1TokenBridge
+L1_TOKEN_BRIDGE_PROXY: str = "0x76943C0D61395d8F2edF9060e1533529cAe05dE6"
+L1_TOKEN_BRIDGE_NEW_IMPL: str = "0xc4E3ff0b5B106f88Fc64c43031BE8b076ee9F21C"
+
+# 2. L2TokenBridge
+L2_TOKEN_BRIDGE_PROXY: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+L2_TOKEN_BRIDGE_NEW_IMPL: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+
+# 3. wstETH on L2
+L2_NON_REBASABLE_TOKEN_PROXY: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+L2_NON_REBASABLE_TOKEN_NEW_IMPL: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+
+# 4. stETH on L2
+L2_REBASABLE_TOKEN_PROXY: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+L2_REBASABLE_TOKEN_NEW_IMPL: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+
+# 5. TokenRateOracle on L2
+L2_TOKEN_RATE_ORACLE_PROXY: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+L2_TOKEN_RATE_ORACLE_NEW_IMPL: str = "0x6b314986E3737Ce23c2a13036e77b3f5A846F8AF"
+
+def encode_upgrade_call(proxy_admin: str, proxy: str, new_impl: str):
+    scroll_executor = interface.OpBridgeExecutor(
+        "0x6625c6332c9f91f2d27c304e729b86db87a3f504"
+    )  # any address to bypass
+
+    params = eth_abi.encode(["address", "address"], [proxy, new_impl])
+
+    return scroll_executor.queue.encode_input([proxy_admin], [0], ["upgrade(address,address)"], [params], [False])
+
+def encode_l1_l2_sendMessage(to: str, fee: int, calldata: str):
+    l1_l2_msg_service = interface.OpCrossDomainMessenger(OPTIMISM_L1_L2_MESSAGE_SERVICE)
+
+    return l1_l2_msg_service.sendMessage.encode_input(to, fee, calldata, 1_000_000)
+
+def start_vote(tx_params: Dict[str, str], silent: bool) -> bool | list[int | TransactionReceipt | None]:
```

Comment:
Overall it can't be that simple because L1 actions would be executed immediately at the moment of the vote enactment, while L2 actions would be postponed further due to bridging delays (~15-30 minutes, including additional manual execution of the queued actions on the Optimism's side).

Therefore, we need to iron up the upgrade plan to prevent any hiccups and tackle asynchronous upgrades and ACL complications.